bundles/org.openhab.binding.nuki/src/main/java/org/openhab/binding/nuki/internal/discovery/NukiBridgeDiscoveryService.java

```diff
@@ -0,0 +1,168 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.nuki.internal.discovery;
+
+import java.util.Collections;
+import java.util.concurrent.ScheduledExecutorService;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.http.HttpStatus;
+import org.openhab.binding.nuki.internal.constants.NukiBindingConstants;
+import org.openhab.binding.nuki.internal.constants.NukiLinkBuilder;
+import org.openhab.binding.nuki.internal.dto.BridgeApiAuthDto;
+import org.openhab.binding.nuki.internal.dto.WebApiBridgeDiscoveryDto;
+import org.openhab.binding.nuki.internal.dto.WebApiBridgeDto;
+import org.openhab.core.common.ThreadPoolManager;
+import org.openhab.core.config.discovery.AbstractDiscoveryService;
+import org.openhab.core.config.discovery.DiscoveryResult;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.config.discovery.DiscoveryService;
+import org.openhab.core.id.InstanceUUID;
+import org.openhab.core.io.net.http.HttpClientFactory;
+import org.openhab.core.thing.ThingUID;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * Discovery service which uses Nuki Web API to discover all bridges on same network
+ * and uses authentication API to obtain access token.
+ *
+ * @author Jan Vybíral - Initial contribution
+ */
+@NonNullByDefault
+@Component(service = DiscoveryService.class, configurationPid = "discovery." + NukiBindingConstants.BINDING_ID)
+public class NukiBridgeDiscoveryService extends AbstractDiscoveryService {
+
+    private final Logger logger = LoggerFactory.getLogger(NukiBridgeDiscoveryService.class);
+
+    private final HttpClient httpClient;
+    private final Gson gson = new Gson();
+    private final ScheduledExecutorService threadPool = ThreadPoolManager.getScheduledPool("discovery");
+
+    @Activate
+    public NukiBridgeDiscoveryService(@Reference final HttpClientFactory httpClientFactory) {
+        super(Collections.singleton(NukiBindingConstants.THING_TYPE_BRIDGE), 30, false);
+        this.httpClient = httpClientFactory.getCommonHttpClient();
+    }
+
+    @Override
+    protected void startScan() {
+        logger.info("Bridge discovery started");
```

Comment:
Logging to info should be used sparsely e.g. a newly started component or a user file that has been loaded. This could be removed at all, since it is already logged by the framework.